Hippo.Core/Services/AccountUpdateService.cs

```diff
@@ -177,6 +179,44 @@ private async Task<Result> CompleteCreateAccount(QueuedEventDataModel data)
             return Result.Ok();
         }
 
+        private async Task<Result> CompleteCreateGroup(QueuedEventDataModel data)
+        {
+            var accountModel = data.Accounts.FirstOrDefault();
+            var groupModel = data.Groups.FirstOrDefault();
+            var account = await _dbContext.Accounts
+                .Where(a => a.Cluster.Name == data.Cluster && a.Owner.Kerberos == accountModel.Kerberos)
+                .FirstOrDefaultAsync();
+            var groupExists = await _dbContext.Groups
+                .AnyAsync(g => g.Cluster.Name == data.Cluster && g.Name == groupModel.Name);
+
+            if (accountModel == null || groupModel == null)
+            {
+                return Result.Error("Invalid data: action {Action} requires one account and one group", QueuedEvent.Actions.CreateAccount);
+            }
+
+            if (groupExists)
+            {
+                return Result.Error("Group already exists: {Name} on cluster {Cluster}", groupModel.Name, data.Cluster);
+            }
+
+            if (account == null)
+            {
+                return Result.Error("Account does not exist: {Kerberos} on cluster {Cluster}", accountModel.Kerberos, data.Cluster);
+            }
+
+            var newGroup = new Group
+            {
+                Name = groupModel.Name,
+                DisplayName = data.Metadata?["DisplayName"] ?? groupModel.Name,
+                AdminAccounts = new () { account },
+                ClusterId = account.ClusterId
+            };
+            
+
+            await _dbContext.Groups.AddAsync(newGroup);
```

Comment:
Just checking to make sure you don't need to save changes here. Might be doing it elsewhere

Reply:
It's done by the caller. I'll add a comment about it...